tests/integration/cli/test_ssh.py

```diff
@@ -0,0 +1,45 @@
+# Import Python libs
+from __future__ import absolute_import, print_function, unicode_literals
+
+# Import test Libs
+from tests.support.case import SSHCase
+
+
+class SSHCustomModuleTest(SSHCase):
+    '''
+    Test sls with custom module functionality using ssh
+    '''
+    def run_ssh(self, arg_str, with_retcode=False, catch_stderr=False,
+                timeout=60, wipe=True, raw=False):  # pylint: disable=W0221
+        '''
+        Execute salt-ssh
+        '''
+        arg_str = '{0} -ldebug{1} -c {2} -i --ssh-options="PubkeyAuthentication=no" --roster-file {4} --out=json localhost {5}'.format(
+            ' -W' if wipe else '',
+            ' -r' if raw else '',
+            self.get_config_dir(),
+            os.path.join(RUNTIME_VARS.TMP_CONF_DIR, 'key_test'),
+            os.path.join(RUNTIME_VARS.TMP_CONF_DIR, 'roster'),
+            arg_str)
+        ret = self.run_script('salt-ssh',
+                              arg_str,
+                              with_retcode=with_retcode,
+                              catch_stderr=catch_stderr,
+                              timeout=timeout,
+                              raw=True)
+        log.debug('Result of run_ssh for command \'%s\': %s', arg_str, ret)
+        return ret
+
+    def run_function(self, function, arg=(), timeout=90, wipe=True, raw=False, stdin=None, **kwargs):
+        '''
+        We use a 90s timeout here, which some slower systems do end up needing
+        '''
+        ret = self.run_ssh(self._arg_str(function, arg), timeout=timeout,
+                           wipe=wipe, raw=raw, stdin=stdin)
```

Comment:
I'm seeing this error when trying to run the test:

```
   -> tests.integration.cli.test_ssh.SSHCustomModuleTest.test_ssh_bad_passwd  .............................
       Traceback (most recent call last):
         File "/home/ch3ll/git/salt/tests/integration/cli/test_ssh.py", line 45, in test_ssh_bad_passwd
           cmd = self.run_function('test.ping', stdin=['y\n', 'testpass\n'])
         File "/home/ch3ll/git/salt/tests/integration/cli/test_ssh.py", line 38, in run_function
           wipe=wipe, raw=raw, stdin=stdin)
       TypeError: run_ssh() got an unexpected keyword argument 'stdin'
```

Reply:
sorry, those ones shouldn't be in there, that was an old commit i forgot to get rid of before pushing

Reply:
I tried to make it so that we could pass stdin, in and actually simulate the whole thing, but we don't have a good way in any of the `self.run_*` commands in any test case to pass in stdin that I could find, so i went with a unit test of the thing that I was actually modifying.